Parser/pegen/pegen.c

```diff
@@ -530,13 +547,21 @@ _PyPegen_fill_token(Parser *p)
 
     if (p->fill == p->size) {
         int newsize = p->size * 2;
-        p->tokens = PyMem_Realloc(p->tokens, newsize * sizeof(Token *));
-        if (p->tokens == NULL) {
-            PyErr_Format(PyExc_MemoryError, "Realloc tokens failed");
+        Token **new_tokens = PyMem_Realloc(p->tokens, newsize * sizeof(Token *));
+        if (new_tokens == NULL) {
+            PyErr_NoMemory();
             return -1;
         }
+        else {
+            p->tokens = new_tokens;
+        }
         for (int i = p->size; i < newsize; i++) {
             p->tokens[i] = PyMem_Malloc(sizeof(Token));
+            if (p->tokens[i] == NULL) {
+                p->size = i - 1; // Needed, in order to cleanup correctly after parser fails
```

Comment:
Are you use about `i -1`? _PyPegen_Parser_Free():

```
    for (int i = 0; i < p->size; i++) {
        PyMem_Free(p->tokens[i]);
    }
```

For example, if the loop fails at the first iteration, you replace p->size=SIZE with p->size=(SIZE - 1), no?

Reply:
You're very right. I was thinking that the loop in `_PyPegen_Parser_Free` was going up to and including `p->size` for some reason. Pushing a fix.